### DISCUSSION TITLE
IPOPT never gets to first iteration

### DISCUSSION THREAD

**Discussion by dhathris** [coin-or/Ipopt#786] [Q&A]
I have a C++ CasADi IPOPT solver program with custom callbacks for both the objective and constraints with enable_fd set to true. The solver seems to be stuck in preprocessing without ever reaching the first IPOPT iteration. MUMPS is the linear solver. What could be the potential root causes of this issue?

**Comment by svigerske** [comment]
Probably CasADI doing finite difference evaluation for derivatives. You may want to try a small version of your NLP and look into the timing statistics of Ipopt to see where the time is spent. It may also be a bit faster to enable the Hessian approximation in Ipopt. But most of all, don't use finite differences for derivatives, but let CasADI do automatic differentiation or provide derivatives.
